helix-view/src/theme.rs

```diff
@@ -42,17 +45,41 @@ impl Loader {
         if name == "base16_default" {
             return Ok(self.base16_default());
         }
-        let filename = format!("{}.toml", name);
 
-        let user_path = self.user_dir.join(&filename);
-        let path = if user_path.exists() {
-            user_path
+        let theme_toml = self.load_theme(name, name, false)?;
+
+        Ok(Theme::from(theme_toml))
+    }
+
+    // load the theme and its parent recursively and merge them
+    // `base_theme_name` is the theme from the config.toml,
+    // used to prevent some circular loading scenarios
+    fn load_theme(
+        &self,
+        name: &str,
+        base_them_name: &str,
+        only_default_dir: bool,
+    ) -> Result<Value, anyhow::Error> {
+        let path = self.path(name, only_default_dir);
+        let theme_toml = self.load_toml(path)?;
+
+        let inherits_from = theme_toml.get("inherits_from");
+
+        let theme_toml = if let Some(parent_theme_name) = inherits_from {
+            let parent_theme_name = parent_theme_name.to_string().replace('\"', "");
+
+            let parent_theme_toml = self.load_theme(
+                &parent_theme_name,
+                base_them_name,
+                base_them_name == parent_theme_name,
+            )?;
```

Comment:
```suggestion
            let parent_theme_name = parent_theme_name.as_str().ok_or(anyhow!(
                "Theme: expected 'inherits' to be a string: {}",
                parent_theme_name
            ))?;

            let parent_theme_toml = self.load_theme(
                parent_theme_name,
                base_them_name,
                base_them_name == parent_theme_name,
            )?;
```

The `"`s are because this is a `toml::Value`. We'll want to check that it's parsed as a string and fail if it isn't (for example, a theme provides a list of strings of theme names to inherit).

Reply:
Thanks I somehow missed the `as_str` method, still a far way to get Rust into my system.

But clippy suggested to use `ok_or_else(|| anyhow!{...}` instead of `ok_or` which I did, since the pipeline checks with clippy.